scripts/message/MessageForm.js

```diff
@@ -13,34 +13,79 @@ to save messages sent through the form
-// declare and export MessageForm function to build HTML for message form
-
-import { getChosenUser, setView, saveMessage , fetchUsers} from "../data/TransientState.js"
-
-const chosenUsers = getChosenUser()
-// const savedMessages = saveMessage() This function should only be called in the event listener -Ryan
-// const preferredView = setView() This function should only be called in the event listener -Ryan
-
-export const MessageForm = async () => {
-    const users = await fetchUsers()
-
-    let html = `<h1>Create Your Message</h1> 
-   <section>
-   To:
-    <select id="messageRecipientDropdown"> 
-    <option value="0">List of Users`
-    for (const user of users) {
-      html +=  `<option value = "${user.id}">${user.name}</option>`
-    }
-
-return html += `</select></section>
-<section>
-<label for="username">Name:</label><br>
-<input type="text" id="name"><br>
-<label for="text">Subject:</label><br>
-<input type="text" id="subject"><br>
-<label for="message">Message:</label><br>
-<input type="text" id="message"><br>
-<button type="button">Send</button>
-</section>
-` }
+
+        
+
+
+
+        import { getChosenUser, setMessage, setView, saveMessage , fetchUsers, saveLike, getCurrentUser} from "../data/TransientState.js"
+
+
+
+        // const documentingSetMessage = (event) => {
+        //     if(clickEvent.target.id === "send") {
+        //     setMessage(parseInt(clickEvent.target.value))
+        // }
+        // }
+        const sendMessageButtonLink = async (clickEvent) => {
+             if(clickEvent.target.id === "sendMessageForm") {
+                let message = document.getElementById("messageForm").value
+                let currentUser = getCurrentUser()
+                let recipientId = document.getElementById("chooseRecipient").value
+        
+                let messageObject = {
+                "userId": currentUser.userId,
+                "recipientId": recipientId,
+                "read": false,
+                "text": message
+            }        
+                setMessage(messageObject)    
+                await saveMessage()
+                setView(defaultView)
+            }
+
+            else if(clickEvent.target.id === "cancelMessageForm") {
+                setView(defaultView)
+                const customEvent = new CustomEvent("stateChanged")
+                document.dispatchEvent(customEvent)
+            }
+        
+        }
+        
+        
+           
+        
+                   
+        
+        
+        
+        
+        export const MessageForm = async () => {
+            const users = await fetchUsers()
+        
+            let html = `<h1>Create Your Message</h1> 
+           <section>
+           To:
+            <select id="messageRecipientDropdown"> 
+            <option value="0">List of Users`
+            for (const user of users) {
+              html +=  `<option value = "${user.id}" id = "chooseRecipient" >${user.name}</option>`
```

Comment:
You can get rid of the 'id' for each 'option', because the value is what we really want, and id's should be unique for each element. Also, I think people usually leave no space between the equals signs when assigning attributes for HTML elements, but it probably doesn't really make a difference.